runtime/src/bank/partitioned_epoch_rewards/calculation.rs

```diff
@@ -336,10 +324,21 @@ impl Bank {
 
                     let stake_pubkey = **stake_pubkey;
                     let vote_pubkey = stake_account.delegation().voter_pubkey;
-                    let vote_account = get_vote_account(&vote_pubkey)?;
-                    if vote_account.owner() != &solana_vote_program {
-                        return None;
+                    let vote_account_from_cache = cached_vote_accounts.get(&vote_pubkey);
+                    if ASSERT_STAKE_CACHE && vote_account_from_cache.is_none() {
+                        let account_from_db = self.get_account_with_fixed_root(&vote_pubkey);
+                        if let Some(account_from_db) = account_from_db {
+                            if account_from_db.owner() == &solana_vote_program::id()
+                                && VoteState::deserialize(account_from_db.data()).is_ok()
```

Comment:
I prefer if we keep this check consistent with the stakes cache `check_and_store` check:

```rust
            if VoteStateVersions::is_correct_size_and_initialized(account.data()) {
                match VoteAccount::try_from(account.to_account_shared_data()) {
```

note that `VoteAccount::try_from` does an owner check too

Reply:
Yes. you are right. 
updated.